In [4]:
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
import sklearn 
import pandas as pd
import keras
import tensorflow as tf
from sklearn.metrics import ConfusionMatrixDisplay, confusion_matrix
from sklearn.preprocessing import MinMaxScaler

2024-06-21 18:50:09.097882: I external/local_tsl/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-21 18:50:09.101361: I external/local_tsl/tsl/cuda/cudart_stub.cc:32] Could not find cuda drivers on your machine, GPU will not be used.
2024-06-21 18:50:09.148881: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-06-21 18:50:10.055714: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT


leer los datos del dataset

In [29]:
data_file = pd.read_csv("/home/nicolas/Github/Programacion/machileLearning/tensorflow_clasificadores/heart_attack/heart.csv")

con data_file.head() te muestra los primeros 5 datos esto normalmente se realiza para poder visualizar las columnas y estar familiarizado con los datos

In [30]:
print(data_file.head())

   age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  output  
0    0      1       1  
1    0      2       1  
2    0      2       1  
3    0      2       1  
4    0      2       1  


age       :Edad

sex       :Sexo del paciente donde: 1 -> para hombre y 0 -> para mujer

cp        :Chest pain type -> indica el nivel del dolor en el pecho del paciente

trtbps    :Resting Blood Pressure: la presion de paciente en reposo(en mm de 
           Hg(mercurio)), al entrar al hospital

chol      :Colestero serico: colesterol del paciente en mg/dl

fbs       :Fasting blood sugar: si el azucar del paciente en ayunas es mayor a 120 mg/dl
           (1 = True, 0 = False)

restecg   :Resting Electrocardiographic Results: representa los resultados del 
           electrocardiograma con 1 como normal, o 0 como alguna anomalia 

thalachh  :frecuencia cardiaca maxima alcanzada, por el paciente
  
exgn      :Excercise Induced Angina -> angina inducida por el ejercicio 1 = si, 0 = no
  
oldpeak   :ST Depression induced by exercise Relative to rest: 
           La depresión del ST inducida por el ejercicio en relación con el reposo, 
           una medida de la salud del corazó durante el ejercicio.
  
slp       :The Slope of the Peak Exercise ST Segment : La pendiente del segmento 
           ST del ejercicio máximo, con valores que indican la naturaleza de la 
           pendiente durante el ejercicio máximo.
  
caa       :Number of Major Vessels : el número de vasos principales (0-3) coloreados
           mediante fluoroscopia.
  
thall     :Thalassemia : Un trastorno sanguíneo llamado talasemia, representado 
           como 3 = normal; 6 = defecto fijo; 7 = defecto reversible. La codificación 
           puede variar.
  
output    :salida : La variable objetivo que indica la presencia de enfermedad 
           cardíaca (1) o ausencia (0).

In [26]:
print(type(data_file))

<class 'pandas.core.frame.DataFrame'>


In [31]:
target_column = "output"
numerical_column = data_file.columns.drop(target_column)
output_rows = data_file[target_column]
data_file.drop(target_column,axis=1,inplace=True)


In [34]:
print(data_file.head())
input_rows = data_file[numerical_column]
print(input_rows)


   age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0   63    1   3     145   233    1        0       150     0      2.3    0   
1   37    1   2     130   250    0        1       187     0      3.5    0   
2   41    0   1     130   204    0        0       172     0      1.4    2   
3   56    1   1     120   236    0        1       178     0      0.8    2   
4   57    0   0     120   354    0        1       163     1      0.6    2   

   caa  thall  
0    0      1  
1    0      2  
2    0      2  
3    0      2  
4    0      2  
     age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
0     63    1   3     145   233    1        0       150     0      2.3    0   
1     37    1   2     130   250    0        1       187     0      3.5    0   
2     41    0   1     130   204    0        0       172     0      1.4    2   
3     56    1   1     120   236    0        1       178     0      0.8    2   
4     57    0   0     120   354    0        1 